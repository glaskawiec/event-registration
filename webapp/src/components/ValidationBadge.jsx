import React from 'react';
import PropTypes from 'prop-types';
import { Badge } from 'react-bootstrap';

const ValidationBadge = props => (
    (props.touched &&
    props.error &&
    <Badge variant="danger">
        {props.error}
    </Badge>) || null
)

ValidationBadge.propTypes = {
    touched: PropTypes.bool,
    error: PropTypes.string,
};

export default ValidationBadge;