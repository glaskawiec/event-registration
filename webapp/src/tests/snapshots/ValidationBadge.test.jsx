import React from 'react';
import renderer from 'react-test-renderer';
import ValidationBadge from '../../components/ValidationBadge';

test('ValidationBadge snapshot', () => {
    const props = {
        touched: false,
        error: ''
    };
    const tree = renderer
        .create(
            <ValidationBadge {...props} />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});