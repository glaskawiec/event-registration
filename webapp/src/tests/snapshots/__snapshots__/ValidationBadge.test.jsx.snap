// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationBadge snapshot 1`] = `null`;
